src/App.tsx

```diff
@@ -1,93 +1,136 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { TodosFilter } from './components/TodosFilter';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { SortBy, Todo } from './types';
+
+let todos = localStorage.getItem('todosStorage');
+let todosFromLocalStorage: Todo[] = [];
+
+if (todos !== null) {
+  todosFromLocalStorage = JSON.parse(todos);
+  if (!Array.isArray(todosFromLocalStorage)) {
+    todosFromLocalStorage = [];
+  }
+}
 
 export const App: React.FC = () => {
-  return (
-    <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+  const [completedTodosId, setCompletedTodosId] = useState<number[]>(
+    todosFromLocalStorage.filter(todo => todo.completed)
+      .map(todoId => todoId.id),
+  );
+  const [todosList, setTodos] = useState<Todo[]>(
+    todosFromLocalStorage,
+  );
+  const [sortBy, setSortBy] = useState<SortBy>(SortBy.all);
+  const [todosLeft, setTodosLeft] = useState<number>(
+    todosList.length - completedTodosId.length,
+  );
```

Comment:
why do you need to have the state for todosLeft if it's just a calculated value?